### PR TITLE
[CVE-2020-36632] Bumps flat from 4.1.1 to 5.0.2

### PR DESCRIPTION
**Pull request by ZilongX**
Signed-off-by: Zilong Xia <zilongx@amazon.com>

### Description
* Bump package `flat` from `4.1.1` to `5.0.2` to resolve [CVE-2020-36632](https://github.com/advisories/GHSA-2j2x-2gpw-g8fm) in branch `2.x` (release candidate for `OSD 2.6.0`)
* `flat` is introduced into OSD by `mocha@^7.2.0`
* `CVE-2020-36632` is fixed from `mocha@8.2.0` (https://github.com/mochajs/mocha/blob/v8.2.0/package-lock.json)
* Yet since `mocha 8.0.0` brings in a bunch of breaking changes (https://github.com/mochajs/mocha/blob/master/CHANGELOG.md), leveraging `resolutions` here instead of the major version bumping
* `mocha` has been bumped up to `10.1.0` in `main` branch btw (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2711), which would be released as part of `OSD 3.0.0`
* No breaking changes identified for the `flat` version bump (https://github.com/hughsk/flat/commits/master?before=7a184e7adbe06f2acafab15a9efef667cc452abb+35&branch=master&qualified_name=refs%2Fheads%2Fmaster)
* Actual CVE fix changes - https://github.com/hughsk/flat/commit/20ef0ef55dfa028caddaedbcb33efbdb04d18e13
 
### Issues Resolved
* Resolved https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3167
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by seanneumann — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### 🛡 Security
 - [CVE-2022-37601][CVE-2022-37599] Bump loader-utils to 2.0.4 ([#3318](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3318))
 - [CVE-2022-25860] Bumps simple-git from 3.15.1 to 3.16.0 ([#3345](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3345))
+- [CVE-2020-36632] Bumps flat from 4.1.1 to 5.0.2 ([#3419](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3419))
```
Comment (joshuarrrr) [inline CHANGELOG.md:11]:
```suggestion
- [CVE-2020-36632] [REQUIRES PLUGIN VALIDATION] Bumps flat from 4.1.1 to 5.0.2 ([#3419](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3419)). To the best of our knowledge, this is a non-breaking change, but if your plugin relies on `mocha` tests, validate that they still work correctly (and plan to migrate them to `jest` [in preparation for `mocha` deprecation](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1572). 
```

**Review by joshuarrrr — COMMENTED** [review]
@ZilongX Thanks for the detailed description.

I think this is probably a better link for looking at the commit diff: https://github.com/hughsk/flat/compare/4.1.1...5.0.2

I have no concerns about this change for OSD core, because we've already removed all mocha tests via https://github.com/opensearch-project/OpenSearch-Dashboards/issues/215 (and still just have a lingering item to remove the scaffolding and dependency: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1572)

But because of that we don't have a great way of validating that any plugins with `mocha` tests configured will be unaffected by this change. So I think we should do an extra callout in the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (ZilongX) [reply CHANGELOG.md:11]: Just to confirm have all mocha tests been removed from both `main` (targeting 3.0) and `2.x` (targeting 2.6) ?

Comment (ZilongX) [reply CHANGELOG.md:11]: And for CHANGELOG didn't realize we can now add call-outs ~ feels like more like a release note thing but it looks good to me

Comment (joshuarrrr) [reply CHANGELOG.md]: > Just to confirm have all mocha tests been removed from both main (targeting 3.0) and 2.x (targeting 2.6) ?

Yep! https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1678

> And for CHANGELOG didn't realize we can now add call-outs ~ feels like more like a release note thing but it looks good to me

Yeah, the idea is for the CHANGELOG to be the initial draft of the release notes. So this is just my suggestion of how we might go about flagging such a callout 🤷

**Review by AMoo-Miki — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Comment by joshuarrrr** [comment]
DCO check failed on a merge commit - will be fixed by squash and merge

**Comment by joshuarrrr** [comment]
@ZilongX Any plans to backport to `1.x`/`1.3`?

**Comment by ZilongX** [comment]
@joshuarrrr , yes backporting to 1.x/1.3 may need some manual efforts though, I'll take a look later. (maybe just 1.3 since there won't be any new minor version release s for 1.x ~)

**Pull request merged**